### DISCUSSION TITLE
ThirdReality Zigbee Water Leak Sensors Firmware

### DISCUSSION THREAD

**Discussion by ahahamyan** [Koenkk/zigbee2mqtt#14840] [Devices]
I purchased 8 of the ThirdReality Zigbee Water Leak Sensors -  with Dripping Detection - https://www.amazon.com/gp/product/B09GYFN8VL  and use them with HomeAssistant / Zigbee2MQTT. The devices are well supported [https://www.zigbee2mqtt.io/devices/3RWS18BZ.html#third%2520reality-3rws18bz](https://www.zigbee2mqtt.io/devices/3RWS18BZ.html#third%20reality-3rws18bz) and work great. I noticed that a few of the sensors are not reporting battery percentage correctly and they are showing over 100 percent, such as 165% battery, etc.. The ones that show percentage correctly (less than 100%) are running firmware v1.00.23  and the ones that show percentage incorrectly (over 100) are running firmware v1.00.24.  Zigbee2MQTT on HomeAssistant provides OTA capabilities https://www.zigbee2mqtt.io/guide/usage/ota_updates.html#automatic-checking-for-available-updates but it seems ThirdReality OTA firmware files are not available?

Perhaps this issue was fixed in later firmware for these. Where can I find the latest firmware for these and update (without needing to use other hub such as the ThirdReality Hub and mobile app)?

**Comment by ahahamyan** [comment]
I reached out 3rdReality support and they confirmed that the issue with battery reporting was fixed in latest firmware. Workaround with 1.0.0.24 is to divide the number by two to get accurate battery percentage remaining. They mentioned that they are working on providing OTA support in near future for updating firmware.

**Reply by JoeCotellese** [reply]
Thanks, just came here to find this information. My sensors indicate an OTA update is available but when I try to do the update it fails with a 403 error.

Couldn't find in the source code where Z2M looks for firmware files.

**Reply by ahahamyan** [reply]
This was made available between Nov 23 - Nov 28 as Third Reality added the OTA support and I updated my devices just fine using Z2M from HA. From my understanding the Z2M OTA index file is at https://github.com/Koenkk/zigbee-OTA/blob/master/index.json and the following section below there defines where the OTA file is (this entry is for the water leak sensor, there are other entries there for other Third Reality Products). However it appears the access to the third reality S3 bucket where the OTA file is denied currently (it was not before) - https://tr-zha.s3.amazonaws.com/firmwares/TRTL_WaterLeak_DEV_V31_v1.00.31.ota  - this is the history of when Third Reality checked in the changes - https://github.com/Koenkk/zigbee-OTA/commits?author=weihuan1111 - I would think they would need to fix the public access issue on the AWS S3 Bucket.

 {
        "fileVersion": 49,
        "fileSize": 145746,
        "manufacturerCode": 4659,
        "imageType": 54179,
        "sha512": "929637894934b7aede459920cc1d6ef377a2fee9bc03237bc867cd0037647e09a5e52b1f04e5ff3bf0a73666002dc22cf4fb2c690729c4d5cf04178f3f6fb6c5",
        "url": "https://tr-zha.s3.amazonaws.com/firmwares/TRTL_WaterLeak_DEV_V31_v1.00.31.ota"
    },

**Comment by mathmaniac43** [comment]
I have two of these sensors reporting firmware version v1.00.34 in Z2M. In HA, they report that they are version 34, and that version 52 is available.

I can't figure out how to update them. I press the update button in Z2M, but after a few minutes get an error that the device did not respond to the update request. Is there a trick to ensure battery powered ZigBee devices like these check with Z2M and allow the update to occur?

**Reply by shad282** [reply]
I need help with this too and should we update them to 52?
